src/contextmenu.js

```diff
@@ -0,0 +1,472 @@
+          selectedBlocksLength + ')';
+        }
+      } else {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['INLINE_INPUTS'];
+        } else {
+          return Blockly.Msg['INLINE_INPUTS'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!block.isInFlyout && block.isMovable() && !block.isCollapsed()) {
+        for (let i = 1; i < block.inputList.length; i++) {
+          // Only display this option if there are two value or dummy inputs
+          // next to each other.
+          if (block.inputList[i - 1].type !== Blockly.inputTypes.STATEMENT &&
+            block.inputList[i].type !== Blockly.inputTypes.STATEMENT) {
+            return 'enabled';
+          }
+        }
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
+      const state = !scope.block.getInputsInline();
+      const apply = function(block) {
+        if (block && inlineOption.preconditionFn({
+          block: block,
+        }) === 'enabled') {
+          block.setInputsInline(state);
+        }
+      };
+      Blockly.Events.setGroup(true);
+      if (scope.block && blockSelection.size === 0) {
+        apply(scope.block);
+      }
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        if (block) {
+          apply(block);
+        }
+      });
+      Blockly.Events.setGroup(false);
+    },
+    scopeType: Blockly.ContextMenuRegistry.ScopeType.BLOCK,
+    id: 'blockInline',
+    weight: 3,
+  };
+  Blockly.ContextMenuRegistry.registry.register(inlineOption);
+};
+
+/**
+ * Modification for context menu 'Collapse/Expand' to be available for
+ * multiple blocks.
+ */
+const registerCollapseExpandBlock = function() {
+  const collapseExpandOption = {
+    displayText: function(scope) {
+      const selectedBlocksLength = blockSelection.size;
+      if (scope.block.isCollapsed()) {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['EXPAND_BLOCK'];
+        } else {
+          return Blockly.Msg['EXPAND_BLOCK'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      } else {
+        if (selectedBlocksLength <= 1) {
+          return Blockly.Msg['COLLAPSE_BLOCK'];
+        } else {
+          return Blockly.Msg['COLLAPSE_BLOCK'] + ' (' +
+          selectedBlocksLength + ')';
+        }
+      }
+    },
+    preconditionFn: function(scope) {
+      const block = scope.block;
+      if (!block.isInFlyout && block.isMovable() &&
+        block.workspace.options.collapse) {
+        return 'enabled';
+      }
+      return 'hidden';
+    },
+    callback: function(scope) {
+      const state = !scope.block.isCollapsed();
+      const apply = function(block) {
+        if (block && collapseExpandOption.preconditionFn({
+          block: block,
+        }) === 'enabled' && !hasSelectedParent(block)) {
```

Comment:
Is it possible to select a block that is a child of a collapsed block? If so, you may want to expand a collapsed block even if it does have a selected parent. Although, you don't want to collapse selected blocks that have selected parents.

Reply:
Cool, I didn't even realize that, thanks!